File: CS/Contacts/Controls/PreviewControl.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraEditors;
using DevExpress.XtraReports.UI;
using Contacts.Reports;

namespace Contacts.Controls
{
    public partial class PreviewControl : XtraUserControl, IControl
    {
        private string _url;
        public PreviewControl(string url = null)
        {
            InitializeComponent();
            _url = url;
        }
        

        public int Id { get; set; }

        public bool IsDirty
        {
            get { return true; }
        }

        public void Save()
        {
            return;
        }

        public void Reload()
        {
            if (_url != null)
            {
                var report = XtraReport.FromFile(_url, true);
                if (report is IReport)
                    (report as IReport).LoadDataSource();
                printControl1.PrintingSystem =
                    report.PrintingSystem;
                report.CreateDocument();
            }
        }

        public int Delete()
        {
            return -1;
        }

        public TabType TabType
        {
            get { return TabType.Preview; }
        }

    }
}

File: VB/Contacts/Controls/PreviewControl.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Drawing
Imports System.Data
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraEditors
Imports DevExpress.XtraReports.UI
Imports Contacts.Reports

Namespace Contacts.Controls
	Partial Public Class PreviewControl
		Inherits XtraUserControl
		Implements IControl
		Private _url As String
		Public Sub New(Optional ByVal url As String = Nothing)
			InitializeComponent()
			_url = url
		End Sub


		Private privateId As Integer
		Public Property Id() As Integer Implements IControl.Id
			Get
				Return privateId
			End Get
			Set(ByVal value As Integer)
				privateId = value
			End Set
		End Property

		Public ReadOnly Property IsDirty() As Boolean Implements IControl.IsDirty
			Get
				Return True
			End Get
		End Property

		Public Sub Save() Implements IControl.Save
			Return
		End Sub

		Public Sub Reload() Implements IControl.Reload
			If _url IsNot Nothing Then
				Dim report = XtraReport.FromFile(_url, True)
				If TypeOf report Is IReport Then
					TryCast(report, IReport).LoadDataSource()
				End If
				printControl1.PrintingSystem = report.PrintingSystem
				report.CreateDocument()
			End If
		End Sub

		Public Function Delete() As Integer Implements IControl.Delete
			Return -1
		End Function

		Public ReadOnly Property TabType() As TabType Implements IControl.TabType
			Get
				Return TabType.Preview
			End Get
		End Property

	End Class
End Namespace
